core/src/main/java/org/apache/iceberg/io/FileAccessFactory.java

```diff
@@ -0,0 +1,139 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.io;
+
+import org.apache.iceberg.FileFormat;
+import org.apache.iceberg.data.Record;
+import org.apache.iceberg.deletes.PositionDelete;
+
+/**
+ * Interface that provides a unified abstraction for converting between data file formats and
+ * input/output data representations.
+ *
+ * <p>FileAccessFactory serves as a bridge between storage formats ({@link FileFormat}) and expected
+ * input/output data structures, optimizing performance through direct conversion without
+ * intermediate representations. File format implementations handle the low-level parsing details
+ * while the object model determines the in-memory representation used for the parsed data.
+ * Together, these provide a consistent API for consuming data files while optimizing for specific
+ * processing engines.
+ *
+ * <p>Iceberg provides these built-in object models:
+ *
+ * <ul>
+ *   <li><strong>generic</strong> - for {@link Record} objects (engine-agnostic)
+ *   <li><strong>spark</strong> - for Apache Spark InternalRow objects
+ *   <li><strong>spark-vectorized</strong> - for columnar batch processing in Spark (not supported
+ *       with {@link FileFormat#AVRO})
+ *   <li><strong>flink</strong> - for Apache Flink RowData objects
+ *   <li><strong>arrow</strong> - for Apache Arrow columnar format (only supported with {@link
+ *       FileFormat#PARQUET})
+ * </ul>
+ *
+ * <p>Processing engines can implement custom object models to integrate with Iceberg's file reading
+ * and writing capabilities.
+ *
+ * @param <E> input schema type used when converting input data to the file format
+ * @param <D> output type used for reading data, and input type for writing data and deletes
+ */
+public interface FileAccessFactory<E, D> {
```

Comment:
It's a best practice to avoid the name `Factory` unless the class is actually used as a factory, meaning that it is created with configuration and then used to create objects using that configuration similar to how the `FileWriterFactory` is used. In addition, it isn't clear what "access" means here and I would expect the class to have something to do with _accessing_ files rather than creating builders for a specific object model.

I see that there was [a thread](https://github.com/apache/iceberg/pull/12774#discussion_r2093626096) about using `ObjectModel` and this was renamed. I understand the criticism of `ObjectModel`, but I think that the rename adds to the confusion instead of making it more understandable.

Reply:
Reverted anyway